api/objects.proto

```diff
@@ -122,6 +122,13 @@ message Service {
 	// UpdateStatus contains the status of an update, if one is in
 	// progress.
 	UpdateStatus update_status = 5;
+
+	// MarkedRemoved indicates whether a service has been marked for
+	// removal. This happens when a service rm comes in from the
+	// API, and the orchestrator is waiting for all the tasks to
+	// be properly shut down before removing the service object
+	// from the store.
+	bool marked_removed = 7;
```

Comment:
I hate to nit pick on this, but marked_removed is a little confusing. Maybe marked_for_removal or pending_removal ?

Reply:
I don't mind updating it. Maybe let's wait to see if others have any comments, then I can update it once at the end.